### ISSUE TITLE
Null checking fails when using MemberNotNullWhen and property patterns

### ISSUE THREAD

**Issue by skarllot** [dotnet/roslyn-analyzers#4487]
### Analyzer

**Diagnostic ID**: CS8600 and CS8603

### Describe the bug

The analyzer fails to check nullability using `MemberNotNullWhenAttribute` and property patterns.

### Steps To Reproduce

The following code triggers the analyzer warning.

```csharp
#nullable enable

public struct Test
{
    [MemberNotNullWhen(true, nameof(Value))]
    public bool IsOk { get; }
    public bool IsError { get; }
    public string? Value { get; }
}

public class Program {
    public string Main(Test test) {
        switch (test){
            case { IsOk: true }:
                return test.Value;    // <-- Possible null reference return
            case { IsError: true }:
                return String.Empty;
            default:
                return String.Empty;
        }
    }
}
```

### Expected behavior

Follows the attributes and check correctly.

## Additional context

The code above does not trigger the analyzer warning.

```csharp
public class Program {
    public string Main(Test test) {
        switch (test){
            case { } when test.IsOk:
                return test.Value;    // <-- No error here
            case { } when test.IsError:
                return String.Empty;
            default:
                return String.Empty;
        }
    }
}
```

The example code can be found on Sharplab: [link](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEBMBGAsAKBQBgAIUMA6AEQEsIBzAOwHtZKBjKUgYQYBM4BBOhAA2ATyiUoAbly4AxHQCuQoRGBC4hOILVwZeAMyFYCBSxiEAKnFi4A3rkKPCAbQCycMMEQA5BjG9KQgDqABZaABQwJnAANISCYHAMAGbhAGrCCnAAlNkAug5OKIbADAxChACSUADyANaEtoQ0cDCShAC+hY7FhKXlVVAAoggIDAiNza3tXThOxIYk+AD8hBlCWZMtbZ24s7i96MQYAOyN3QvHRFawGOE35vCw2edz805QAO6UMCwhhJFrDBsvY3u93ixoBomtV6iBCFFNh0QBdweCUGcnjBSOssu0nAB6AmEAA8AFoyYQAApMcQ6eKBQgIODJRBaFgaDGotGQqDQwYjMYIeGIjTI7loopnADKUUodBopCGYAADjARNIwZLCLxkhAlDAUVrtVLCLKEPLFcq1RqJbstbN5hdDhhrkC0PcgQigS9QZKvj8/gCsSC7fNefyOoRPmE6N7YKRYXUjSb0ZigTjMnB8Y4iaSKYRfJpRuNCGFmWGnBHJlGY1p49jqoLxinU/MMWa5Qqlar1Zq27r9UJDZW052Ld3rX27Y6nLN9jhcAlrCqIBzjhRqPQmDBWOwuLwBMIxBI7BdnHwYHLgAp4ABVKC0ODhS/X29wCwQBDbdjuGAhHhCCQQhXwtG94E/b9WnYKkxhVRB1TiSo6HLH44G4QgAF5CD1IQ+TiPhlAYT5XANSgVXULCEWifILnleAEEECo+WEdDiDQQh3E8Hw/ACZRQi0UDKHAjR4SEkSLj9NFen6CoACVWgURjcX5bYZj0SUXXwZw8k4jwvAQKAtmme07V6LiDN8fxAgEuhxPfcJZOIE4VLiJZCESbiEGyO0pJNBSYCUugVKojEVP7VMLMQIzsLoOBPh0yZPIMzoIvBBdNMMKKECsvjgljez4EcsoKjCrM4lXBAIDAIylkS5Lop840nD87UAqCkLsLKjZs1HPSvJijz9OitL3lnRx5yAA===)
